src/main/java/gyro/google/storage/BucketResource.java

```diff
@@ -0,0 +1,625 @@
+ *                      age: 15
+ *                  end
+ *              end
+ *          end
+ *
+ *          logging
+ *              log-bucket: 'test-bucket'
+ *              log-object-prefix: 'gyro'
+ *          end
+ *
+ *          retention-policy
+ *              retention-period: 3300
+ *          end
+ *
+ *          website
+ *              main-page-suffix: 'index.html'
+ *              not-found-page: '404.jpg'
+ *          end
+ *      end
+ */
+@Type("bucket")
+public class BucketResource extends GoogleResource implements Copyable<Bucket> {
+
+    private static final String LABEL_REGEX = "[^a-z0-9_-]";
+    private static final Pattern LABEL_PATTERN = Pattern.compile(LABEL_REGEX);
+
+    private String id;
+    private List<BucketAccessControlConfiguration> acl;
+    private String name;
+    private Map<String, String> labels;
+    private String location;
+    private List<BucketCors> cors;
+    private BucketBilling billing;
+    private Boolean defaultEventBasedHold;
+    private BucketEncryption encryption;
+    private BucketIamConfiguration iamConfiguration;
+    private BucketLifecycle lifecycle;
+    private BucketLogging logging;
+    private BucketRetentionPolicy retentionPolicy;
+    private String storageClass;
+    private BucketVersioning versioning;
+    private BucketWebsite website;
+    private String selfLink;
+
+    /**
+     * The generated ID for the bucket. (Read Only)
+     */
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    /**
+     * Access controls on the bucket. See `Bucket Access Controls <https://cloud.google.com/storage/docs/json_api/v1/bucketAccessControls/>`_.
+     *
+     * @subresource gyro.google.storage.BucketAccessControlConfiguration
+     */
+    @Updatable
+    public List<BucketAccessControlConfiguration> getAcl() {
+        if (acl == null) {
+            return new ArrayList<>();
+        }
+        return acl;
+    }
+
+    public void setAcl(List<BucketAccessControlConfiguration> acl) {
+        this.acl = acl;
+    }
+
+    /**
+     * A unique name for the Bucket conforming to the `bucket naming guidelines <https://cloud.google.com/storage/docs/naming?authuser=1#requirements/>`_.
+     */
+    @Id
+    @Required
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Optional set of up to 64 key:value metadata pairs. Each key:value must conform to `Label guidelines <https://cloud.google.com/storage/docs/key-terms?#bucket-labels/>`_.
+     */
+    @Updatable
+    public Map<String, String> getLabels() {
+        return labels;
+    }
+
+    public void setLabels(Map<String, String> labels) {
+        this.labels = labels;
+    }
+
+    /**
+     * The geographic region objects within the bucket will reside. Default is ``US``. See `Bucket locations <https://cloud.google.com/storage/docs/locations/>`_.
```

Comment:
As we don't set the default in the code, better not to mention that, as depending on api or service principal configuration if the default changes then the doc would reflect the wrong value.

```suggestion
     * The geographic region objects within the bucket will reside.  See `Bucket locations <https://cloud.google.com/storage/docs/locations/>`_.
```

Reply:
Done.